code/ai/aigoals.cpp

```diff
@@ -1150,8 +1151,8 @@ void ai_add_goal_sub_sexp( int sexp, int type, ai_info *aip, ai_goal *aigp, cons
 		}
 	}
 
-	if ( aigp->priority > MAX_GOAL_PRIORITY ) {
-		nprintf (("AI", "bashing sexpression priority of goal %s from %d to %d.\n", CTEXT(node), aigp->priority, MAX_GOAL_PRIORITY));
+	if ( aigp->priority > MAX_GOAL_PRIORITY || priority_is_nan || priority_is_nan_forever ) {
```

Comment:
Assigning a goal with a priority higher than 90 can interfere with the player's ability to issue orders. This is rarely, but not never, done intentionally. With this change, it's possible for it to now happen _unintentionally,_ which seems like a footgun in the making. How'd you feel about having it issue a debug warning and cancel the order?

Reply:
Good points.  I'll do that.

(Aside: when `eval_num` encounters SEXP_NAN, it returns 0 and sets the flags, so `aigp->priority` would be 0 here.)